### DISCUSSION TITLE
eza v0.11.0

### DISCUSSION THREAD

**Discussion by cafkafk** [eza-community/eza#191] [Announcements]
# Description

A new minor release, and the first minor release in the lifetime of eza. Why a minor release? Because we just landed windows support, that's why!

A lot of code quality changes in this release as well, of particular note, the icons are now stored in a perfect hash table, giving a small but noticeable improvement in performance. 

# Changelog

### Bug Fixes

- Add windows implementation of is_empty_dir
- Re-align `--git-ignore` in help message
- Avoid direnv error if nix isn't installed

### Documentation

- Empty dir functions
- Document is_empty_dir functions
- Add function documentation for get_file_type and icon_for_file.

### Features

- Optimize checking for empty directories when a directory has subdirectories
- Use perfect hash tables for file types and icons

### Miscellaneous Tasks

- Bump git2 from 0.17.2 to 0.18.0
- Bump uzers from 0.11.1 to 0.11.2
- Bump glob from 0.3.0 to 0.3.1
- Bump DeterminateSystems/nix-installer-action from 3 to 4
- Bump terminal_size from 0.1.16 to 0.2.6
- Bump timeago from 0.3.1 to 0.4.1

### Refactor

- Use phf macros instead of codegen to create icon and filetype tables
- Add constants for most of the commonly used icons
- Add constants for the rest of icons used multiple times
- Rename class FileExtension to FileTypeClassifier to better reflect the purpose
- Move get_file_type to FileType enum

### Styling

- Is_empty_dir() was put between the unix size() and windows size()

### Build

- Use rust stable
- Add unstable package
- Disable clippy check 'unreadable_literal' in generated files

## New Contributors
* @ariasuni made their first contribution in https://github.com/eza-community/eza/pull/188
* @cfxegbert made their first contribution in https://github.com/eza-community/eza/pull/183
* @daviessm made their first contribution in https://github.com/eza-community/eza/pull/171

**Full Changelog**: https://github.com/eza-community/eza/compare/v0.10.9...v0.11.0

<hr /><em>This discussion was created from the release <a href='https://github.com/eza-community/eza/releases/tag/v0.11.0'>eza v0.11.0</a>.</em>

**Comment by gierens** [comment]
So I've updated the deb repo as well ... but I had to alter our dependencies a little, see #192

**Comment by OmiChrome** [comment]
Release - Binary when?

**Reply by cafkafk** [reply]
As soon as I have time to finish up #101, but I'm not comfortable giving specific deadlines.

**Reply by cafkafk** [reply]
#211 makes this possible for future releases.

> Release - Binary when?

Just now. And likely Monday releases will now have release binaries.

Currently, we create binaries for:
- x86_64-unknown-linux-gnu
- aarch64-unknown-linux-gnu
- x86_64-pc-windows-gnu

I've added them to https://github.com/eza-community/eza/releases/tag/v0.11.0 and would love testing.
